UMLClass.py

```diff
@@ -0,0 +1,111 @@
+'''
+Samantha Noggle
+
+Adds a class, deletes a class, & renames a class
+'''
+
+# List of all class objects the user has created
+classIndex = []
+
+class UMLClass:
+    def __init__(self, name: str):
+        self.name = name
+        self.attributes = []
+
+    def rename(self, newName):
+        print(f"\nClass \"{self.name}\" was renamed to \"{newName}\"")
+        self.name = newName
+
+
+def isNameUnique(name: str):
+    '''
+    Checks classIndex for duplicate names
+    '''
+    for c in classIndex:
+        if c.name == name:
+            return False
+    return True
+
+
+def findClass(name: str):
+    """
+    Finds a class object by name and returns it's index
+    """
+    for i, c in enumerate(classIndex):
```

Comment:
why do you use two iterators for one list?

Reply:
It looks like c is the class object and i keeps track of the index

Reply:
what do you mean? i keeps track of which index we are at, c holds the value of the index